tests/backup/backup_license_test.go

```diff
@@ -105,36 +110,208 @@ var _ = Describe("{NodeCountForLicensing}", func() {
-		nodeLabels, err = core.Instance().GetLabelsOnNode(sourceClusterWorkerNodes[0].Name)
-		if err != nil {
-			dash.VerifySafely(err, nil, fmt.Sprintf("Getting label from worker node %v", sourceClusterWorkerNodes[0].Name))
+		dash.VerifySafely(err, nil, "Switching context to source cluster")
+		log.InfoD("Removing label portworx.io/nobackup=true from all worker nodes on source cluster if present")
+		for _, workerNode := range sourceClusterWorkerNodes {
+			err = RemoveLabelFromNodesIfPresent(workerNode, "portworx.io/nobackup")
+			dash.VerifySafely(err, nil, fmt.Sprintf("Removing label portworx.io/nobackup=true from worker node %s", workerNode))
+		}
+		CleanupCloudSettingsAndClusters(nil, "", "", ctx)
+	})
+})
+
+// LicensingCountWithNodeLabelledBeforeClusterAddition applies label portworx.io/nobackup=true before adding application cluster to backup and verifies the license count
+var _ = Describe("{LicensingCountWithNodeLabelledBeforeClusterAddition}", func() {
+	var (
+		cloudCredName                 string
+		cloudCredUID                  string
+		bkpLocationName               string
+		backupLocationUID             string
+		backupName                    string
+		clusterUid                    string
+		restoreName                   string
+		bkpNamespaces                 []string
+		sourceClusterWorkerNodes      []node.Node
+		destinationClusterWorkerNodes []node.Node
+		totalNumberOfWorkerNodes      []node.Node
+		contexts                      []*scheduler.Context
+		appContexts                   []*scheduler.Context
+	)
+	backupLocationMap := make(map[string]string)
+	JustBeforeEach(func() {
+		StartTorpedoTest("NodeCountForLicensingBeforeClusterAddition",
+			"Applies label portworx.io/nobackup=true before adding of application cluster to backup and verifies the license count", nil, 82954)
+		log.InfoD("Deploy applications needed for taking backup")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
 		}
-		for key := range nodeLabels {
-			if key == "portworx.io/nobackup" {
-				log.InfoD("Removing the applied label portworx.io/nobackup=true from worker nodes on source cluster at the end of the testcase")
-				err = Inst().S.RemoveLabelOnNode(sourceClusterWorkerNodes[0], "portworx.io/nobackup")
-				dash.VerifySafely(err, nil, fmt.Sprintf("Removing label portworx.io/nobackup=true from worker node %v", sourceClusterWorkerNodes[0].Name))
-				break
+	})
+
+	It("Label the application cluster nodes before adding to backup and verify the license count", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
```

Comment:
How about incorporating it in every step instead of making it a global variable?

Reply:
As discussed, keeping it as global variable only

Reply:
Makes sense.